docs/content/release-notes/0.3.0.md

```diff
@@ -42,17 +61,83 @@ switch "test-switch" {
 
 ```
 
-Documentation was updated in [\#371](https://github.com/asteris-llc/converge/pull/371) so that links like
-[converge.aster.is/0.2.0](https://converge.aster.is/0.2.0) will resolve documentation for that version.
+#### Named Locks
+
+Basic locks have been added to the engine. This allows worksflows where multiple tasks run
+one at a time instead of in parallel. Locks are added as a node in the graph and released
+after all tasks sharing the lock are complete.
+
+For example, `apt` package installs cannot run in parallel. In the following example
+each task with the `apt` lock is run sequentually.
+
+```hcl
+task "install-jq" {
+  check = "dpkg -s jq >/dev/null 2>&1"
+  apply = "apt-get update 2>&1 > /dev/null && apt-get -y install jq"
+  lock  = "apt"
+}
+
+task "install-build-essential" {
+  check = "dpkg -s build-essential >/dev/null 2>&1"
+  apply = "apt-get update 2>&1 > /dev/null && apt-get -y install build-essential"
+  lock  = "apt"
+}
+```
+
+#### Node Metadata
+
+Initial support for adding metadata to graph nodes was added via [\#369](https://github.com/asteris-llc/converge/pull/369). 
+
+#### Conditional Waits
+
+Support for waiting for a task or port was [\#334](https://github.com/asteris-llc/converge/pull/334). A user
+can now structure workflows where tasks can wait for a port to be open or a condition to be met.
+
+Below is an example of waiting for a port to be open using the `wait.port` resource:
+
+```hcl
+wait.port "8080" {
+  host         = "localhost"
+  port         = 8080
+  interval     = "1s"
+  max_retry    = 10
+  grace_period = "2s"
+}
+```
+
+The following shows using `wait.query` to wait using a task:
+
+```hcl
+wait.query "service-health" {
+  check        = "nc -z localhost 8080"
+  interval     = "1s"
+  max_retry    = 10
+  grace_period = "1s"
+}
+```
+
+Wait is documented at: [wait.port]({{< relref "resources/wait.port.md" >}}) and [wait.query]({{< relref "resources/wait.query.md" >}}).
+
+#### Output Improvements
+
+Text output for changes is now aligned via [\#317](https://github.com/asteris-llc/converge/pull/317).
 
 ## Bug Fixes
 
-This releases fixes several race condition (#266) and ordering (#254) bugs that would cause Converge to error out of otherwise valid executions.
+- docker.container regression [\#343](https://github.com/asteris-llc/converge/issues/343)
+- handle parameters with valid zero value [\#338](https://github.com/asteris-llc/converge/issues/338)
+- shell module should not set StatusLevel based on process exit code [\#323](https://github.com/asteris-llc/converge/issues/323)
+- param dependency fails in `samples/shellContext.hcl` [\#313](https://github.com/asteris-llc/converge/issues/313)
+- Execution Engine Ignoring Warning Levels [\#243](https://github.com/asteris-llc/converge/issues/243)
+- Make pipeline functions use mult-return instead of Either [\#225](https://github.com/asteris-llc/converge/issues/225)
 
-## Examples
+## Examples/Documentation
```

Comment:
passive voice should be fixed in this section

...

I mean, paragraphs in this section should be in the active voice.